systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java

```diff
@@ -0,0 +1,417 @@
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBasicExampleAdminClient;
+import io.strimzi.systemtest.resources.operator.SetupClusterOperator;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@Tag(REGRESSION)
+public class ThrottlingQuotaST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(TopicST.class);
+    static final String NAMESPACE = "throttling-quota-cluster-test";
+    final String kafkaUsername = "test-quota-user";
+    final String classTopicPrefix = "quota-topic-test";
+
+    /**
+     * Test checks for throttling quotas set for user
+     * on creation & deletion of topics and create partition operations.
+     */
+    @ParallelNamespaceTest
+    @Tag(INTERNAL_CLIENTS_USED)
+    void testThrottlingQuotasCreateTopic(ExtensionContext extensionContext) {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(NAMESPACE, extensionContext);
+        int topicsCountOverQuota = 500;
+        String createAdminName = "create-admin-client";
+        String topicNamePrefix = classTopicPrefix + "-create";
+
+        setupKafkaInNamespace(extensionContext, clusterName, namespaceName);
+
+        KafkaBasicExampleAdminClient adminClientJob = new KafkaBasicExampleAdminClient.Builder()
+                .withAdminName(createAdminName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicNamePrefix)
+                .withTopicCount(topicsCountOverQuota)
+                .withNamespaceName(namespaceName)
+                .withTopicOperation("create")
```

Comment:
Do we want some enums for these kind of operations? I think it would be more clearer and everyone will also know all the operation they can use.

Reply:
I was thinking about it. But not at test level. Might be a good idea.. Need to put it in some reasonable location though...